src/decoder.ts

```diff
@@ -383,6 +382,72 @@ export class Decoder<A> {
     });
   }
 
+  /**
+   * Decoder for fixed-length arrays, aka Tuples.
+   *
+   * Supports up to 8-tuples.
+   *
+   * Example:
+   * ```
+   * tuple([number(), number(), string()]).run([5, 10, 'px'])
+   * // => {ok: true, result: [5, 10, 'px']}
+   * ```
+   */
+  static tuple<A>(decoder: [Decoder<A>]): Decoder<[A]>;
+  static tuple<A, B>(decoder: [Decoder<A>, Decoder<B>]): Decoder<[A, B]>;
+  static tuple<A, B, C>(decoder: [Decoder<A>, Decoder<B>, Decoder<C>]): Decoder<[A, B, C]>;
+  static tuple<A, B, C, D>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>]
+  ): Decoder<[A, B, C, D]>;
```

Comment:
nit: feel free to slap a `// prettier-ignore` at the end of each of these lines to prevent wrapping, like with the `union` decoder. There isn't much human relevant info in these defs, so I think it's all-around easier to parse as single lines.

Reply:
Agreed.